code/espurna/wifi.ino

```diff
@@ -32,12 +32,23 @@ void _wifiCheckAP() {
 
 void _wifiConfigure() {
 
-    jw.setHostname(getSetting("hostname").c_str());
+    const String hostname = getSetting("hostname");
+    if (!inRangeInc(hostname.length(), 1u, 31u)) {
+        DEBUG_MSG_P(PSTR("[WIFI] Warning: \"hostname\" must be 1..31 characters\n"));
```

Comment:
True. I'll do checking routine separately and use it on boot and here.

Reply:
Done inside getHostname / getAdminPass wrapping getSetting(...) and with setSetting inside to avoid failure on next call.
wsOnAfterParse aka module config reloader is a nice place to put it otherwise (right at the start of callbacks chain), but calling things like `wifi.reset` individually would also require running all of validation before actually configuring wifi module...